src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java

```diff
@@ -0,0 +1,505 @@
+        this.mContext = context;
+        this.mBuilder = builder;
+        this.mDrawableIds = getResourceIds(context);
+        this.mNow = now;
+    }
+
+    /* package */ Notification createNotification(Intent inboundIntent) {
+        parseIntent(inboundIntent);
+
+        if (this.mData == null) {
+            return null;
+        }
+
+        if (this.mData.isSilent()) {
+            MPLog.i(LOGTAG, "Notification will not be shown because \'mp_silent = true\'");
+            return null;
+        }
+
+        buildNotificationFromData();
+
+        Notification n;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            n = mBuilder.build();
+        } else {
+            n = mBuilder.getNotification();
+        }
+
+        if (!mData.isSticky()) {
+            n.flags |= Notification.FLAG_AUTO_CANCEL;
+        }
+
+        return n;
+    }
+
+    /* package */ void parseIntent(Intent inboundIntent) {
+        List<MixpanelNotificationData.MixpanelNotificationButtonData> buttons = new ArrayList<>();
+        final String message = inboundIntent.getStringExtra("mp_message");
+        final String iconName = inboundIntent.getStringExtra("mp_icnm");
+        final String largeIconName = inboundIntent.getStringExtra("mp_icnm_l");
+        final String whiteIconName = inboundIntent.getStringExtra("mp_icnm_w");
+        final String expandableImageURL = inboundIntent.getStringExtra("mp_img");
+        final String uriString = inboundIntent.getStringExtra("mp_cta");
+        CharSequence notificationTitle = inboundIntent.getStringExtra("mp_title");
+        CharSequence notificationSubText = inboundIntent.getStringExtra("mp_subtxt");
+        final String colorName = inboundIntent.getStringExtra("mp_color");
+        final String buttonsJsonStr = inboundIntent.getStringExtra("mp_buttons");
+        final String campaignId = inboundIntent.getStringExtra("mp_campaign_id");
+        final String messageId = inboundIntent.getStringExtra("mp_message_id");
+        final String extraLogData = inboundIntent.getStringExtra("mp");
+        final int badgeCount = inboundIntent.getIntExtra("mp_bdgcnt", MixpanelNotificationData.NOT_SET);
+        final String channelId = inboundIntent.getStringExtra("mp_channel_id");
+        final String notificationTag = inboundIntent.getStringExtra("mp_tag");
+        final String groupKey = inboundIntent.getStringExtra("mp_groupkey");
+        final String ticker = inboundIntent.getStringExtra("mp_ticker");
+        final String stickyString = inboundIntent.getStringExtra("mp_sticky");
+        final String timeString = inboundIntent.getStringExtra("mp_time");
+        final int visibility = inboundIntent.getIntExtra("mp_visibility", Notification.VISIBILITY_PRIVATE);
+        final String silent = inboundIntent.getStringExtra("mp_silent");
+
+        trackCampaignReceived(campaignId, messageId, extraLogData);
+
+        if (message == null) {
```

Comment:
a push notification with no body is still shown - are we sending empty strings? we need to think how we are handling silent push with no body